dockstore-common/src/test/java/io/dockstore/common/yaml/DockstoreYamlTest.java

```diff
@@ -198,4 +202,81 @@ public void testGalaxySubclass() throws DockstoreYamlHelper.DockstoreYamlExcepti
         assertEquals(4, workflows.stream().filter(w -> w.getSubclass().equalsIgnoreCase("gxformat2")).count());
     }
 
+    @Test
+    public void testGitReferenceFilter() {
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("anything"), List.of()));
+
+        // Generic name filter
+        // Glob
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/develop"), List.of("develop")));
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/develop"), List.of("develop")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/foo/develop"), List.of("develop")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/foo/develop"), List.of("develop")));
+        // Regex
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/develop"), List.of("/develop/")));
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/develop"), List.of("/develop/")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/foo/develop"), List.of("/develop/")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/foo/develop"), List.of("/develop/")));
+
+        // Branch-only name filter
+        // Glob
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/develop"), List.of("heads/develop")));
```

Comment:
Can continue discussion in #dockstore but we probably need a new thread. 

I apologize for not catching this earlier, I'm not keen on the user needing to know the `heads` syntax, it doesn't seem to show up in the CI docs (I'm referring to circle, travis, appveyer) 

But I think we can simplify things by doing regexes and exact match only, not globs as well (matching CI docs)
Alternatively, we can discuss trade-offs of exact match and globs only?

I realize this contradicts the original ticket. This is a lot hairier than I originally expected.